### ISSUE TITLE
[Improvement] Make tf.contrib.model_pruning.masked_conv2d API compatible with tf.layers.conv2d

### ISSUE THREAD

**Issue by jsiloto** [tensorflow/tensorflow#20702]
TensorFlow version: v1.8.0-0-g93bc2e2072 1.8.0

### Describe the problem
I know contrib isn't officially supported, but I don't know who I should request this to:
masked_conv2d is intended to be a plug-in replacement of conv2d, but argument names are different. Some simples changes such as: num_outputs->filters, scope->name etc should be enough to make it compatible

**Comment by yongtang** [comment]
I remember tf.contrib is always subject to change. Though would like to be confirmed by API review before working on updating the API.

**Comment by asimshankar** [comment]
Correct, we do not promise API stability for symbols in `tf.contrib` so they can be changed.
Removing the "API Review" label.

I'll defer to the maintainers of the `model_pruning` module on whether they want to make the change though.

@suyoggupta

**Comment by tensorflowbutler** [comment]
Nagging Assignee @suyoggupta: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @suyoggupta: It has been 44 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @suyoggupta: It has been 59 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @suyoggupta: It has been 74 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @suyoggupta: It has been 104 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @suyoggupta: It has been 119 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @suyoggupta: It has been 134 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.
